crates/sui-graphql-rpc/src/types/cursor.rs

```diff
@@ -0,0 +1,357 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{fmt, ops::Deref};
+
+use async_graphql::{
+    connection::{CursorType, OpaqueCursor},
+    *,
+};
+use serde::{de::DeserializeOwned, Serialize};
+
+use crate::{config::ServiceConfig, error::Error};
+
+/// Wrap the `OpaqueCursor` type to implement Scalar for it.
+pub(crate) struct Cursor<C>(OpaqueCursor<C>);
+
+/// Connection field parameters parsed into a single type that encodes
+#[derive(Debug)]
+pub(crate) struct Page<C> {
+    after: Option<Cursor<C>>,
+    before: Option<Cursor<C>>,
+    limit: u64,
+    side: Side,
```

Comment:
I am sure you gave it a lot of thought, but I am just having a hard time wrapping my mind around naming it `side`. It's a bit unintutitive but then again, I also do not have a better name.

Reply:
Could call it `end` instead? That aligns better with `prefix` and `suffix`?

Reply:
I'll leave it up to you, just wanted to mention that.

Reply:
Or slice?

Reply:
I ended up calling it `End` with enumerands `Front` and `Back`.  `slice` doesn't sound great to me because it already means something in Rust, and it seems too general -- slices can be arbitrary subsequences, not necessarily prefixes or suffixes.